### PR TITLE
Fix gtest and related warnings

### LINKED ISSUE #106: Fix GoogleTest warnings.
When I use Google test as the test framework, there are following warnings prompt when building the code. This will block building after Warning As Error being enabled.

Severity	Code	Description	Project	File	Line	Suppression State
Warning	C26812	The enum type 'testing::internal::Mutex::StaticConstructorSelector' is unscoped. Prefer 'enum class' over 'enum' (Enum.3).	az	E:\dev\src\cpp\azure-sdk-for-cpp\out\build\x64-Debug\_deps\googletest-src\googletest\include\gtest\internal\gtest-port.h	1379	
Warning	C26812	The enum type 'testing::TestPartResult::Type' is unscoped. Prefer 'enum class' over 'enum' (Enum.3).	az	E:\dev\src\cpp\azure-sdk-for-cpp\out\build\x64-Debug\_deps\googletest-src\googletest\include\gtest\gtest-test-part.h	63	
Warning	C26495	Variable 'testing::internal::Mutex::critical_section_' is uninitialized. Always initialize a member variable (type.6).	az	E:\dev\src\cpp\azure-sdk-for-cpp\out\build\x64-Debug\_deps\googletest-src\googletest\include\gtest\internal\gtest-port.h	1379	
Warning	C26495	Variable 'testing::internal::Mutex::critical_section_init_phase_' is uninitialized. Always initialize a member variable (type.6).	az	E:\dev\src\cpp\azure-sdk-for-cpp\out\build\x64-Debug\_deps\googletest-src\googletest\include\gtest\internal\gtest-port.h	1379	
Warning	C26495	Variable 'testing::internal::Mutex::owner_thread_id_' is uninitialized. Always initialize a member variable (type.6).	az	E:\dev\src\cpp\azure-sdk-for-cpp\out\build\x64-Debug\_deps\googletest-src\googletest\include\gtest\internal\gtest-port.h	1379	
Warning	C26495	Variable 'testing::internal::Mutex::type_' is uninitialized. Always initialize a member variable (type.6).	az	E:\dev\src\cpp\azure-sdk-for-cpp\out\build\x64-Debug\_deps\googletest-src\googletest\include\gtest\internal\gtest-port.h	1379

### PR DESCRIPTION
**Pull request by antkmsft**
Closes #106

### CODE REVIEW

File: sdk/core/azure-core/src/http/curl/curl.cpp [hunk 1]
```diff
@@ -38,11 +38,18 @@ inline void LogThis(std::string const& msg)
 }
 
 template <typename T>
+#if defined(_MSC_VER)
+#pragma warning(push)
+#pragma warning(disable : 26812)
```
Comment (RickWinter) [inline sdk/core/azure-core/src/http/curl/curl.cpp:43]: Add a comment for what warning 26812 is.  Brief description and link to warning documentation for msvc.  Helps for future understanding when reading code.

File: sdk/core/azure-core/src/http/curl/curl_connection_private.hpp [hunk 2]
```diff
@@ -99,7 +99,14 @@ namespace Azure { namespace Core { namespace Http {
       // Get the socket that libcurl is using from handle. Will use this to wait while
       // reading/writing
       // into wire
+#if defined(_MSC_VER)
+#pragma warning(push)
+#pragma warning(disable : 26812)
```
Comment (RickWinter) [inline sdk/core/azure-core/src/http/curl/curl_connection_private.hpp:104]: Ditto from above comment

File: sdk/core/azure-core/test/ut/CMakeLists.txt [hunk 3]
```diff
@@ -59,6 +59,10 @@ target_include_directories (azure-core-test PRIVATE $<BUILD_INTERFACE:${CMAKE_CU
 
 target_link_libraries(azure-core-test PRIVATE azure-core gtest gmock)
 
+if (MSVC)
+  target_compile_options(azure-core-test PUBLIC /wd26495 /wd26812)
```
Comment (RickWinter) [inline sdk/core/azure-core/test/ut/CMakeLists.txt:63]: Ditto from above.  Its good to have comments about the specific warnings we turn on/off.  Helps users to quickly understand what our build system is doing.

**Review by RickWinter — APPROVED** [review]
Please add comments about the warnings before commit.

File: sdk/core/azure-core/src/http/curl/curl.cpp (hunk 1 above)
Comment (ahsonkhan) [reply sdk/core/azure-core/src/http/curl/curl.cpp:43]: Why are we suppressing these warnings in the source code? I thought the issue was specific to tests. Should we fix the source instead?

**Review by vhvb1989 — APPROVED** [review]

File: sdk/core/azure-core/src/http/curl/curl.cpp (hunk 1 above)
Comment (antkmsft) [reply sdk/core/azure-core/src/http/curl/curl.cpp:43]: I just went further and fixed this warning for our source as well. There were only two instances. The warning says, `CURLcode` (defined by CURL library )is a C-style `enum`. Use `enum class`. We don't have control over that, so we fix in the code.

Comment (ahsonkhan) [reply sdk/core/azure-core/src/http/curl/curl.cpp:44]: Makes sense, thanks!

**Pull request merged**